packages/tpetra/core/src/Tpetra_Import_Util2.hpp

```diff
@@ -238,9 +266,11 @@ class SortCrsEntries {
             if (ind_(sk+m) >= ind_(sk)) {
               break;
             }
-            const scalar_type dtemp = val_(sk+m);
-            val_(sk+m)   = val_(sk);
-            val_(sk)     = dtemp;
+            if (val_.dimension_0()) {
```

Comment:
See above; it might be easier for the compiler to optimize if you were to lift this test out of the inner loop.